src/components/CarInfo.component.tsx

```diff
@@ -0,0 +1,160 @@
+import { useMutation } from "@apollo/client";
+import React, { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import styled from "styled-components";
+
+import { UserContext } from "../contexts/UserContext";
+import { User_Cars } from "../graphql/generated/graphql";
+import { CREATE_USER_CAR, DELETE_USER_CAR } from "../graphql/mutations";
+import { deleteCarVariables } from "../graphql/variables";
+import { CarRowInfo } from "../interfaces/Car";
+import { defaultTheme } from "../theme";
+import { PATHNAME } from "../utils";
+import { FavoriteButton } from "./FavoriteButton.component";
+import { FlexColumn, FlexRow, H4, P } from "./styled";
+
+export const CarInfo = ({ img, car }: { img: string; car: CarRowInfo }) => {
+  const { loggedUser } = useContext(UserContext);
+  const [isFav, setIsFavorite] = useState<boolean>(car.isFavorite ?? false);
+  const [createUserCar, { loading: favLoading }] = useMutation(CREATE_USER_CAR);
+  const [deleteUserCar, { loading: unfavLoading }] =
+    useMutation(DELETE_USER_CAR);
+
+  const now = new Date();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setIsFavorite(!!car.isFavorite);
+  }, [car.isFavorite]);
+
+  async function toggleFavorite() {
+    if (!loggedUser) {
+      return navigate(PATHNAME.LOGIN);
+    }
+    try {
+      if (isFav) {
+        await deleteUserCar({
+          variables: deleteCarVariables(loggedUser.id, car.id!),
```

Comment:
question, why do you have a function as variables?

Reply:
the reason why I'm using functions as variables is to take some logic out to another file, I personally think that gives both files readability, but let me know If you think I'm wrong.